pkg/extension/sumologicextension/extension.go

```diff
@@ -268,6 +279,62 @@ func (pm *sumologicExtension) register(ctx context.Context, collectorName string
 	return nil
 }
 
+func (pm *sumologicExtension) heartbeat(ctx context.Context) {
+	const heartbeatUrl = "api/v1/collector/heartbeat"
+	if pm.registrationInfo.CollectorCredentialId == "" || pm.registrationInfo.CollectorId == "" {
+		pm.logger.Error("Collector not registered, cannot send heartbeat")
+		return
+	}
+	u, err := url.Parse(pm.baseUrl + heartbeatUrl)
+	if err != nil {
+		pm.logger.Error("Unable to parse heartbeat URL ", zap.String("error: ", err.Error()))
+		return
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), nil)
+	if err != nil {
+		pm.logger.Error("Unable to create HTTP request. ", zap.String("error: ", err.Error()))
+		return
+	}
+
+	pm.addCollectorCredentials(req)
+	pm.addJSONHeaders(req)
+
+	pm.logger.Info("Heartbeat heartbeat API initialized. Start sending requests to ",
+		zap.String("URL", u.String()),
+	)
```

Comment:
```suggestion
	pm.logger.Info("Heartbeat heartbeat API initialized. Starting sending hearbeat requests",
		zap.String("URL", u.String()),
	)
```

This won't look good IMHO hence not reference the fields in the log message.